src/edu/csus/ecs/pc2/core/util/IJSONTool.java

```diff
@@ -0,0 +1,273 @@
+        String id = "" + account.getClientId().getClientNumber();
+        return id;
+    }
+
+    /**
+     * Returns the problem shortname, if present, otherwise the elementid
+     *
+     * @param problem
+     * @return string which is the shortname or elementid
+     */
+    public static String getProblemId(Problem problem) {
+        String id = problem.getElementId().toString();
+        // if we have a problem shortName use it, otherwise default to the internal id
+        String shortName = problem.getShortName();
+        if (shortName != null && !shortName.equals("")) {
+            id = shortName;
+        }
+        return id;
+    }
+
+    /**
+     * Returns the clarification id which is currently the elementid
+     * Someday change to an ordinal?
+     *
+     * @param clarification
+     * @return string which is the id to use for this clarification
+     */
+    public static String getClarificationId(Clarification clar) {
+        String id = clar.getElementId().toString();
+        return id;
+    }
+
+    /**
+     * Returns the clarification answer id which is currently the elementid
+     * Someday change to an ordinal?
+     *
+     * @param clarAns
+     * @return string which is the id to use for this clarification answer
+     */
+    public static String getClarificationAnswerId(ClarificationAnswer clarAns) {
+        String id = clarAns.getElementId().toString();
+        return id;
+    }
+
+    /**
+     * Return wall time for input elapsed time in ms.
+     *
+     * Calculates based on elapsed time plus contest start time
+     *
+     * @param contest
+     * @param elapsedMS
+     *            - elapsed ms when submission submitted
+     * @return wall time for run.
+     */
+    public static Calendar calculateElapsedWalltime(IInternalContest contest, long elapsedMS) {
+
+        ContestTime time = contest.getContestTime();
+        if (time.getElapsedMins() > 0) {
+
+            Calendar contestStart = time.getContestStartTime();
+
+            long ms = contestStart.getTimeInMillis();
+
+            ms += elapsedMS; // add elapsed time
+
+            // create wall time.
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTimeZone(TimeZone.getTimeZone("GMT"));
+            calendar.setTimeInMillis(ms);
+            return calendar;
+
+        } else {
+            return null;
+        }
+
+    }
+
+    /**
+     * Get submission unique ID
+     *
+     * @param the submission's Run object
+     * @return string representation of the run number
+     */
+    public static String getSubmissionId(Run submission) {
+        return Integer.toString(submission.getNumber());
+    }
+
+    /**
+     * Get judgement type (acronym).
+     */
+    public static String getJudgementType(Judgement judgement) {
+        return judgement.getAcronym();
+    }
+
+    /**
+     * Get languague unique ID
+     * @param language
+     * @return string with the id (eg. 'java', 'cpp', etc.)
```

Comment:
It seems like this comment should also explain what is returned if the language id is empty.  The examples ("java", "cpp") imply that it will _always_ return a human-readable string that "looks like" a language name.

Reply:
Cleared up the comment in next push.